### PR TITLE
express api

### PR DESCRIPTION
**Pull request by Josephineoderland**
## Netlify link
https://master--reliable-kleicha-adc2e5.netlify.app
https://project-express-api-lwx5.onrender.com/

### CODE REVIEW

**Comment by AntonellaMorittu** [comment]
Hi Josephine, please add the link to your deployed API in the readme or in the PR template :)

**Review by AntonellaMorittu — CHANGES_REQUESTED** [review]
Please add the link to your deployed api 👍

File: backend/routes/api.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        filteredSales = filteredSales.filter(
+            (sale) => new Date(sale.date) >= filterDate,
+        )
+    }
+
+    const totalPages = Math.ceil(avocadoSalesData.length / limit)
+
+    const nextPage =
+        page < totalPages
+            ? `http://localhost:3000/sales?page=${page + 1}&limit=${limit}`
```
Comment (viktoria-olo) [inline backend/routes/api.js:52]: The localhost url should probably be updated to the deployed render address.

File: backend/routes/api.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+})
+
+router.get("/sales/:id", (req, res) => {
+    const id = parseInt(req.params.id, 10)
+
+    if (isNaN(id)) {
+        res.status(400).json({
+            error: true,
+            message: "Invalid request",
+            details: "Please provide a valid ID. Example: /sales/1",
```
Comment (viktoria-olo) [inline backend/routes/api.js:206]: Very nice and clear error handling! Same for status 404 below.

**Review by viktoria-olo — COMMENTED** [review]
Great use of endpoints and sorting. Very good error handling. Also nice that you did a frontend part. Well done!

**Review by AntonellaMorittu — APPROVED** [review]

**Pull request closed without merging**